vint.go

```diff
@@ -60,3 +60,38 @@ func mustReadByte(r io.Reader) (byte, error) {
 
 	return b, err
 }
+
+func WriteVInt(w io.Writer, n int64) error {
+	if n >= -112 && n <= 127 {
+		_, err := w.Write([]byte{byte(int8(n))})
+		return err
+	}
+
+	length := -112
+	if n < 0 {
+		n = n ^ -1
+		length = -120
+	}
+
+	tmp := n
+	for tmp != 0 {
+		tmp = tmp >> 8
+		length--
+	}
+
+	lengthByte := byte(int8(length))
+
+	if length < -120 {
+		length = -(length + 120)
+	} else {
+		length = -(length + 112)
+	}
+
+	b := make([]byte, length + 1)
+	b[0] = lengthByte
+	for i := 1; i <= length; i++ {
+	    b[i] = byte(n >> uint(8 * (length - i)))
```

Comment:
specifically:
`./vint.go:93: invalid operation: n >> 8 * (length - i) (mismatched types int64 and int)`